InteractionBase/src/commands/CAddNodeToViewByName.cpp

```diff
@@ -40,18 +41,42 @@ bool CAddNodeToViewByName::canInterpret(Visualization::Item*, Visualization::Ite
 	return commandTokens.size() >= 1 && commandTokens.first() == name();
 }
 
-CommandResult* CAddNodeToViewByName::execute(Visualization::Item*, Visualization::Item* target,
-		const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>&)
+CommandResult* CAddNodeToViewByName::execute(Visualization::Item* source, Visualization::Item* target,
+		const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>& cursor)
 {
 	if (commandTokens.size() < 2)
 		return new CommandResult(new CommandError("Please specify a node to add"));
 
+	auto currentView = target->scene()->currentViewItem();
+	QPoint posToInsert;
+	if (source == currentView && isHorizontal(cursor->region()))
+	{
+		auto below =  findItemBelow(currentView, cursor->region());
```

Comment:
OK I know this has all been very confusing. Let me try to summarize:

To properly handle the case where the view item itself has focus (what you have currently in your code):
* use `dynamic_cast<LayoutCursor*>(cursor)->x()/y()/index()` and `cursor->type()` instead of your custom methods.

To properly handle the case where one of the items inside the view is selected, and the user just happend to invoke the add new view command from there (a case which you currently don't handle, but would be nice):
* use `dynamic_cast<GridLayoutFormElement*> (currentView->currentForm(currentView))` to get the x and y coordinates of the currently selected item. In this case you shouldn't care what the cursor type is.